web_src/js/features/common-global.js

```diff
@@ -302,51 +302,55 @@ async function linkAction(e) {
 }
 
 export function initGlobalLinkActions() {
-  function showDeletePopup(e) {
-    e.preventDefault();
-    const $this = $(this);
-    const dataArray = $this.data();
-    let filter = '';
-    if (this.getAttribute('data-modal-id')) {
-      filter += `#${this.getAttribute('data-modal-id')}`;
-    }
+  for (const el of document.querySelectorAll('.delete-button')) {
+    el.addEventListener('click', (e) => {
+      e.preventDefault();
+      const btn = e.currentTarget;
 
-    const $dialog = $(`.delete.modal${filter}`);
-    $dialog.find('.name').text($this.data('name'));
-    for (const [key, value] of Object.entries(dataArray)) {
-      if (key && key.startsWith('data')) {
-        $dialog.find(`.${key}`).text(value);
-      }
-    }
+      // eslint-disable-next-line github/no-dataset -- code depends on the camel-casing
+      const dataObj = btn.dataset;
 
-    $dialog.modal({
-      closable: false,
-      onApprove: async () => {
-        if ($this.data('type') === 'form') {
-          $($this.data('form')).trigger('submit');
-          return;
+      const modalId = btn.getAttribute('data-modal-id');
+      const modal = document.querySelector(`.delete.modal${modalId ? `#${modalId}` : ''}`);
+      const modalNameEl = modal.querySelector('.name');
+      if (modalNameEl) modalNameEl.textContent = btn.getAttribute('name');
```

Comment:
Old `    $dialog.find('.name').text($this.data('name'));`
Now `modalNameEl.textContent = btn.getAttribute('name');` ?

It should be `data-name`?


TBH I don't feel quite safe to mix this refactoring with the `text()` removal.

I would prefer to refactor the `delete-modal` to a clear one first, and add some tests in the devtest to show how it works.

Reply:
Good catch, I guess I must have overlooked this name in testing.

I'm following your earlier suggestion that said to clean code up and test it while doing the jQuery refactors.

Reply:
Or it could have some comments to explain its behaviors. I had added some to the "show-modal":

Then it's easier to understand the code by the comments.

![image](https://github.com/go-gitea/gitea/assets/2114189/4490bc07-89b0-4b61-b916-663f584082b6)